cypress/integration/block_page_test.js

```diff
@@ -0,0 +1,53 @@
+const blockTypeToLabel = {
+  consumer: 'Consumer',
+  producer: 'Producer',
+  producer_consumer: 'Producer & Consumer',
+};
+
+describe('Block page tests', () => {
+  context('no access before login', () => {
+    it('redirects to home', () => {
+      cy.visit('/blocks/blockname');
+      cy.location('pathname').should('eq', '/login');
+    });
+  });
+
+  context('authenticated', () => {
+    beforeEach(() => {
+      cy.fixture('custom_block').as('customBlock');
+      cy.fixture('native_block').as('nativeBlock');
+    });
+
+    it('successfully loads Block page for a custom block', function () {
+      cy.server();
+      cy.route('GET', '/flow/v1/*/blocks/*', '@customBlock');
+      cy.login();
+      cy.visit(`/blocks/${this.customBlock.data.name}`);
+      cy.location('pathname').should('eq', `/blocks/${this.customBlock.data.name}`);
+      cy.get('h2').contains('Block Details');
+    });
+
+    it('displays correct properties for a custom block', function () {
```

Comment:
this test only checks for values without logging in or visiting pages. Maybe you should wrap the login/visit function in a beforeEach

Reply:
Cypress was designed to write tests that build upon previous ones, unlike the common pattern found in unit-testing where you isolate each test as much as possible.
Here you'd like to avoid re-visiting a page if you already have it loaded.
For instance this test builds upon the state left from the previous test and it makes a set of assertions that can be easily grouped and tagged as a single UI behavior to test.
The rationale is that when a test fails, you can read from the logs the specific behavior that failed and immediately recall what could be the cause.